scss/_patterns_notifications.scss

```
@@ -205,43 +217,73 @@ $notification-text-margin-bottom: $spv--large - $spv-nudge;
+    @include vf-icon-info($color: map-get($colors-dark-theme--tinted-borders, information));
   }
 }
 
 // Positive notification styling
 @mixin vf-notifications-positive {
   .p-notification--positive {
     @extend %vf-notification;
-    @include vf-highlight-bar($color-positive, left, true);
-    @include vf-icon-success;
+    @include vf-highlight-bar($colors--theme--border-positive, left, true);
+    @include vf-icon-success($color: map-get($colors-light-theme--tinted-borders, positive));
+  }
+
+  // FIXME: temporary workaround until icons have proper theme support
+  // stylelint-disable-next-line selector-max-type -- base styles can use type selectors
+  body.is-dark .p-notification--positive {
+    @include vf-icon-success(map-get($colors-dark-theme--tinted-borders, positive), $colors--dark-theme--background-default);
   }
 }
 
 // Caution notification styling
 @mixin vf-notifications-caution {
   .p-notification--caution {
     @extend %vf-notification;
-    @include vf-highlight-bar($color-caution, left, true);
-    @include vf-icon-warning;
+    @include vf-highlight-bar($colors--theme--border-caution, left, true);
+    @include vf-icon-warning($color: map-get($colors-light-theme--tinted-borders, caution));
+  }
+
+  // FIXME: temporary workaround until icons have proper theme support
+  // stylelint-disable-next-line selector-max-type -- base styles can use type selectors
+  body.is-dark .p-notification--caution {
+    @include vf-icon-warning(map-get($colors-dark-theme--tinted-borders, caution), $colors--dark-theme--background-default);
   }
 }
 
 // Negative notification styling
 @mixin vf-notifications-negative {
   .p-notification--negative {
     @extend %vf-notification;
-    @include vf-highlight-bar($color-negative, left, true);
-    @include vf-icon-error;
+    @include vf-highlight-bar($colors--theme--border-negative, left, true);
+    @include vf-icon-error($color: map-get($colors-light-theme--tinted-borders, negative));
+  }
+
+  // FIXME: temporary workaround until icons have proper theme support
+  // stylelint-disable-next-line selector-max-type -- base styles can use type selectors
+  body.is-dark .p-notification--negative {
+    @include vf-icon-error(map-get($colors-dark-theme--tinted-borders, negative), $colors--dark-theme--background-default);
   }
 }
 
 // Information notification styling
 @mixin vf-notifications-information {
   .p-notification--information {
     @extend %vf-notification;
-    @include vf-highlight-bar($color-information, left, true);
-    @include vf-icon-info($color-mid-dark);
+    @include vf-highlight-bar($colors--theme--border-information, left, true);
+    @include vf-icon-info($color: map-get($colors-light-theme--tinted-borders, information));
+  }
+
+  // FIXME: temporary workaround until icons have proper theme support
+  // stylelint-disable-next-line selector-max-type -- base styles can use type selectors
+  body.is-dark .p-notification--information {
```

This should include the situation when `is-dark` is set on component itself. Something like that.

This would apply to all other places where only `body.is-dark` changes icons.

```suggestion
  .p-notification--information.is-dark,
  body.is-dark .p-notification--information {
```